perun-openapi/openapi.yml

```diff
@@ -17018,6 +17035,123 @@ paths:
+        - ConsentsManager
+      operationId: getConsentsForUserWithStatus
+      summary: Return Consents of certain status for user with corresponding id
+      parameters:
+        - $ref: '#/components/parameters/userId'
+        - { name: status, description: "consent status", schema: { $ref: '#/components/schemas/ConsentStatus' },  in: query, required: true }
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfConsentResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/consentsManager/getConsentsForUser/id:
+    get:
+      tags:
+        - ConsentsManager
+      operationId: getConsentsForUser
+      summary: Return Consents for user with corresponding id
+      parameters:
+        - $ref: '#/components/parameters/userId'
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfConsentResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/consentsManager/getConsentsForConsentHub/id-s:
+    get:
+      tags:
+        - ConsentsManager
+      operationId: getConsentsForConsentHubWithStatus
+      summary: Return Consents of certain status for ConsentHub with corresponding id
+      parameters:
+        - $ref: '#/components/parameters/id'
+        - { name: status, description: "consent status", schema: { $ref: '#/components/schemas/ConsentStatus' },  in: query, required: true }
+
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfConsentResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/consentsManager/getConsentsForConsentHub/id:
+    get:
+      tags:
+        - ConsentsManager
+      operationId: getConsentsForConsentHub
+      summary: Return Consents for ConsentHub with corresponding id
+      parameters:
+        - $ref: '#/components/parameters/id'
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfConsentResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/consentsManager/getConsentsForUserAndConsentHub:
+    get:
+      tags:
+        - ConsentsManager
+      operationId: getConsentsForUserAndConsentHub
+      summary: Return Consents for specified user in specified consent hub
+      parameters:
+        - $ref: '#/components/parameters/userId'
+        - $ref: '#/components/parameters/id'
+      responses:
+        '200':
+          $ref: '#/components/responses/ListOfConsentResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+
+  /json/consentsManager/getConsentForUserAndConsentHub:
+    get:
+      tags:
+        - ConsentsManager
+      operationId: getConsentForUserAndConsentHub
+      summary: Return Consent object for chosen user in specified consent hub with specified status
+      parameters:
+        - $ref: '#/components/parameters/userId'
+        - $ref: '#/components/parameters/id'
```

Comment:
I suggest defining a new parameter `consentHubId` (similar to userId) and using it here and in other methods as well.

Reply:
Now it needs to be updated in RPC - it expects `id`, but gets `consentHubId`.

Reply:
Ooh OK, I get it now. Fixed